app.js

```diff
@@ -1,53 +1,87 @@
-const FileTextEditor = require("./textEditorService");
-const parsers = require("./parsers");
+const fs = require("fs/promises");
+const { bootstrap } = require("kaholo-plugin-library");
+const { tryCreateRegexFromString, parsePath, pathExists } = require("./helpers");
 
-function createFile(action) {
-  const fileEditor = new FileTextEditor(action.params.PATH);
-  return fileEditor.createFile({
-    content: action.params.CONTENT,
-  });
+async function createFile({
+  PATH: path,
+  CONTENT: content,
+}) {
+  const parsedPath = parsePath(path);
+  try {
+    await fs.writeFile(parsedPath, content);
+  } catch (error) {
+    throw new Error(`Error creating file ${path}: ${error.message || JSON.stringify(error)}`);
+  }
+  return `Created file ${path}.`;
 }
 
-function createVaultFile(action) {
-  return createFile(action);
+async function appendToFile({
+  PATH: path,
+  CONTENT: content,
+  return: returnContent,
+  dontCreate,
+}) {
+  const parsedPath = parsePath(path);
+  if (!pathExists(parsedPath)) {
```

Comment:
Please flatten the statements.
When facing a situation when one of the condition branches would fail, put it under the condition to continue with the happy path. Try to structure the code in such way that there's only one happy path - return statement at the bottom of the function.